lib/template-mixins.js

```diff
@@ -7,19 +7,34 @@ var Hogan = require('hogan.js'),
     _ = require('underscore'),
     moment = require('moment');
 
+var deprecate = require('deprecate');
+
 // This returns a middleware that places mixins against the `res.locals` object.
 //
-// It should be given:
-// - fields: the data needed to generate mixins options, etc.
 // - options:
-//   - viewDirectory: the folder in which templates are found in.
+//   - viewsDirectory: the folder in which templates are found in.
 //   - viewEngine: the type of view, defaults to 'html'.
 //   - sharedTranslationsKey: used to find translations relatively within
 //     the translations.json. Useful for field and button labels.
-module.exports = function (fields, options) {
+//   - translate: a translate function
+//   - fields: static field configuration (overridden by `res.locals.options.fields`)
+module.exports = function (options, deprecated) {
+    var optionsKeys = ['viewsDirectory', 'viewEngine', 'sharedTranslationsKey', 'translate'];
+
+    var deprecateStr = 'hmpo-template-mixins: Passing field config to mixins is deprecated';
+    deprecateStr += '\nField config will be loaded from res.locals.options.fields';
 
-    fields = fields || {};
+    if (arguments.length === 1 && !_.intersection(Object.keys(options), optionsKeys)) {
+        deprecate(deprecateStr);
+        options.fields = options;
+    }
+    if (arguments.length === 2) {
+        deprecate(deprecateStr);
+        deprecated.fields = options;
+        options = deprecated;
+    }
     options = options || {};
+    options.fields = options.fields || {};
```

Comment:
Should we have tests to show that fields can be loaded eitehr through options or res locals?